components/publish/PublishWidget.vue

```diff
@@ -64,7 +64,18 @@ const { editor } = useTiptap({
 })
 
 const characterCount = $computed(() => {
-  let length = stringLength(htmlToText(editor.value?.getHTML() || ''))
+  const text = htmlToText(editor.value?.getHTML() || '')
+
+  let length = stringLength(text)
+
+  // taken from https://github.com/mastodon/mastodon/blob/07f8b4d1b19f734d04e69daeb4c3421ef9767aac/app/lib/text_formatter.rb
+  const linkRegex = /(https?:\/\/(www\.)?|xmpp:)\S+/g
```

Comment:
is this still valid if we have a link in a code block?

Reply:
I don't think it should matter? It's all the same from the mastodon server perspective

Reply:
![Screenshot from 2023-02-06 22-32-38](https://user-images.githubusercontent.com/20907452/217092851-0f834068-fc62-4bb9-969f-fdaf94927580.png)
It appears that an empty code block takes up 8 characters, therefore what's seen in the screenshot should be the correct behavior since 8 + 23 = 31